File: DirectProgramming/Fortran/Jupyter/OpenMP-offload-training/datatransfer/saxpy_func_solution_length.f90
!$omp target map(tofrom: y(1:ARRAY_SIZE)) map(to:x(1:ARRAY_SIZE)) map(from:is_cpu)
is_cpu=omp_is_initial_device();
do i=1,ARRAY_SIZE
        y(i) = a*x(i) + y(i)
end do
!$omp end target

File: DirectProgramming/C++/Jupyter/OpenMP_Offload_C++/datatransfer/saxpy_func_solution_length.cpp
// Add the target pragma with the map clauses here
#pragma omp target map(tofrom: y[0:ARRAY_SIZE]) map(to: x[0:ARRAY_SIZE]) map(from: is_cpu)
{
    is_cpu = omp_is_initial_device();
    for (i = 0; i < ARRAY_SIZE; i++) {
        y[i] = a * x[i] + y[i];
    }
}
